### PR TITLE
compliance: Add support for Go custom compliance checks and implement two for CIS Kubernetes

### PR DESCRIPTION
**Pull request by vboulineau**
### What does this PR do?

Add possibility to have entirely custom Go checks while keeping compliance syntax

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Describe your test plan

Write there any instructions and details you may have to test your PR.

### CODE REVIEW

File: pkg/compliance/report.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+// Report contains the result of a compliance check
+type Report struct {
+	// Data contains arbitrary data linked to check evaluation
+	Data event.Data
+	// Passed defines whether check was successful or not
+	Passed bool
+}
+
+// BuildReportForUnstructured returns default Report for Kubernetes objects
```
Comment (xornivore) [inline pkg/compliance/report.go:21]: Nit: maybe split it to a separate file, it's somewhat specific to Kubernetes and might confuse someone just looking to understand the structure of the `Report` type.

File: pkg/compliance/resource.go [hunk 2]
```diff
@@ -61,21 +64,56 @@ func (r *Resource) Kind() ResourceKind {
...
 	case r.KubeApiserver != nil:
 		return KindKubernetes
+	case r.Custom != nil:
+		return KindCustom
 	default:
 		return KindInvalid
 	}
 }
 
+// Fields & functions available for File
```
Comment (xornivore) [inline pkg/compliance/resource.go:74]: 👍 This is really helpful to have them all listed in one place.

File: pkg/compliance/resource.go [hunk 3]
```diff
@@ -159,7 +217,36 @@ func (a *Audit) Validate() error {
...
+
+	DockerFuncTemplate = "docker.template"
+)
+
 // DockerResource describes a resource from docker daemon
 type DockerResource struct {
 	Kind string `yaml:"kind"`
 }
+
+// Custom is a special resource handlded by dedicated, custom, functions
```
Comment (xornivore) [inline pkg/compliance/resource.go:248]: Typo
```suggestion
// Custom is a special resource handled by a dedicated function.
```

**Review by xornivore — APPROVED** [review]
Looks great 👍 Just a couple of small comments.

File: pkg/compliance/report.go (hunk 1 above)
Comment (vboulineau) [reply pkg/compliance/report.go]: Move to `report_utils.go` though not great name either

**Pull request merged**